pydevices/HtsDevices/cryocon24c.py

```diff
@@ -29,31 +27,27 @@ class CRYOCON24C(MDSplus.Device):
 
      """
 
-    @staticmethod
-    def inputs():
-        return range(ord('a'), ord('d')+1)
-
-    @staticmethod
-    def loops():
-        return range(0, 4)
+    inputs = "abcd"
+    loops = range(1, 4+1)
 
     parts = [
         {'path': ':COMMENT', 'type': 'text', 'options': ('no_write_shot')},
         {'path': ':INIT_ACTION', 'type': 'action', 'valueExpr': "Action(Dispatch('S','INIT',50,None),Method(None,'INIT',head))",'options': ('no_write_shot',)},
         {'path': ':STOP_ACTION', 'type': 'action', 'valueExpr': "Action(Dispatch('S','STOP',50,None),Method(None,'STORE',head))",'options': ('no_write_shot',)},
         {'path': ':RUNNING', 'type': 'numeric', 'options': ('no_write_model')},
-        ]
-
-    for c in inputs.__func__():
-        parts.append({'path': ':INPUT_%c' %(string.upper(chr(c)),), 'type': 'signal', 'options': ('no_write_model', 'write_once',)})
-        parts.append({'path': ':INPUT_%c:SERIAL_NO' %(string.upper(chr(c)),), 'type': 'TEXT', 'options': ('no_write_shot')})
-        parts.append({'path': ':INPUT_%c:CALIBRATION' %(string.upper(chr(c)),), 'type': 'TEXT', 'options': ('no_write_model', 'write_once',)})
-        parts.append({'path': ':INPUT_%c:RESISTENCE' %(string.upper(chr(c)),), 'type': 'signal', 'options': ('no_write_model', 'write_once',)})
-         
-    for i in loops.__func__():
-        parts.append({'path': ':LOOP_%c' %(str(i+1),), 'type': 'signal', 'options': ('no_write_model', 'write_once',)})
-            
-    del c
+    ]
+    for i in inputs.upper():
+        parts.extend([
+            {'path': ':INPUT_%c' % (i,), 'type': 'signal', 'options': ('no_write_model', 'write_once',)},
+            {'path': ':INPUT_%c:SERIAL_NO' % (i,), 'type': 'TEXT', 'options': ('no_write_shot')},
+            {'path': ':INPUT_%c:CALIBRATION' % (i,), 'type': 'TEXT', 'options': ('no_write_model', 'write_once',)},
+            {'path': ':INPUT_%c:RESISTENCE' % (i,), 'type': 'signal', 'options': ('no_write_model', 'write_once',)},
+        ])
+    for i in loops:
+        parts.extend([
+            {'path': ':LOOP_%d' % (i,), 'type': 'signal', 'options': ('no_write_model', 'write_once',)},
```

Comment:
Same question as above

Reply:
never use .format but use % for compatibility